// react-redux-spinner/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "configurablePendingTasksReducer", {
  enumerable: true,
  get: function get() {
    return _reducer.configurablePendingTasksReducer;
  }
});
Object.defineProperty(exports, "pendingTasksReducer", {
  enumerable: true,
  get: function get() {
    return _reducer.pendingTasksReducer;
  }
});
Object.defineProperty(exports, "pendingTask", {
  enumerable: true,
  get: function get() {
    return _reducer.actionKey;
  }
});
Object.defineProperty(exports, "begin", {
  enumerable: true,
  get: function get() {
    return _reducer.begin;
  }
});
Object.defineProperty(exports, "end", {
  enumerable: true,
  get: function get() {
    return _reducer.end;
  }
});
Object.defineProperty(exports, "endAll", {
  enumerable: true,
  get: function get() {
    return _reducer.endAll;
  }
});
Object.defineProperty(exports, "Spinner", {
  enumerable: true,
  get: function get() {
    return _spinner.default;
  }
});

var _reducer = require("./reducer.js");

var _spinner = _interopRequireDefault(require("./spinner.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
