// react-redux-spinner/lib/reducer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.configurablePendingTasksReducer = exports.pendingTasksReducer = exports.endAll = exports.end = exports.begin = exports.actionKey = void 0;
var actionKey = '@@react-redux-spinner/pending-task';
exports.actionKey = actionKey;
var begin = '@@react-redux-spinner/begin';
exports.begin = begin;
var end = '@@react-redux-spinner/end';
exports.end = end;
var endAll = '@@react-redux-spinner/end-all';
exports.endAll = endAll;

var configurablePendingTasksReducer = function configurablePendingTasksReducer() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      _ref$actionKeyPath = _ref.actionKeyPath,
      actionKeyPath = _ref$actionKeyPath === void 0 ? [] : _ref$actionKeyPath;

  return function () {
    var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
    var action = arguments.length > 1 ? arguments[1] : undefined;
    // Reduce the action to find the configured path
    var obj = actionKeyPath.reduce(function (acc, path) {
      return acc[path];
    }, action);

    if (!obj) {
      // Configured path is not included in this action. Nothing to do.
      return state;
    }

    if (obj[actionKey] === begin) {
      return state + 1;
    }

    if (obj[actionKey] === end) {
      if (state - 1 < 0) {
        throw new RangeError("Number of pending tasks decreased below zero. This indicates you have an error in your code. 'end' is called more often than 'begin'.");
      }

      return state - 1;
    }

    if (obj[actionKey] === endAll) {
      return 0;
    }

    return state;
  };
};

exports.configurablePendingTasksReducer = configurablePendingTasksReducer;
var pendingTasksReducer = configurablePendingTasksReducer();
exports.pendingTasksReducer = pendingTasksReducer;
